.vib/grafana-loki/cypress/cypress/integration/grafana_loki_spec.js

```diff
@@ -0,0 +1,71 @@
+/// <reference types="cypress" />
+import body from '../fixtures/logs-body.json';
+
+it('shows build info endpoint', () => {
+  cy.request({
+    method: 'GET',
+    url: '/loki/api/v1/status/buildinfo',
+    form: true,
+  }).as('buildInfo');
+  cy.get('@buildInfo').should((response) => {
+    expect(response.status).to.eq(200);
+    expect(response.body).to.include.all.keys(
+      'version',
+      'revision',
+      'branch',
+      'buildUser',
+      'goVersion'
+    );
+  });
+});
+
+it('allows Loki to get Grafana logs', () => {
+  cy.request({
+    method: 'GET',
+    url: 'loki/api/v1/query?query={app="grafana-loki"}',
+    form: true,
+  }).as('grafanaLogs');
+  cy.get('@grafanaLogs').should((response) => {
+    expect(response.status).to.eq(200);
+  });
+});
+
+it('checks Loki range endpoint', () => {
+  cy.request({
+    method: 'GET',
+    url: 'loki/api/v1/query?query={job="varlogs"}',
+    form: true,
+  }).as('grafanaLogs');
+  cy.get('@grafanaLogs').should((response) => {
+    expect(response.status).to.eq(200);
+    expect(response.body.data.stats.summary).not.to.be.empty;
+    expect(response.body.data.stats).to.include.all.keys(
+      'summary',
+      'querier',
+      'ingester'
+    );
+  });
+});
+
+it('can publish logs to the endpoint', () => {
```

Comment:
Maybe there would also be interesting to check if the logs are really published by retrieving them afterward.

Reply:
I've investigated a lot of queries as we agreed in a separate call but I haven't found a reliable way of doing this which will not introduce flakiness/unpredictability. 
Therefore I applied the alternative which consists in adding a log and a label and checking for that.
Please take a look and let me know if that works for you.